src/main/java/com/conveyal/analysis/datasource/SpatialLayers.java

```diff
@@ -0,0 +1,125 @@
+package com.conveyal.analysis.datasource;
+
+import com.conveyal.analysis.AnalysisServerException;
+
+import com.conveyal.file.FileStorageFormat;
+import com.google.common.collect.Sets;
+import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.io.FilenameUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Utility class with common static methods for validating and processing uploaded spatial data files.
+ */
+public abstract class SpatialLayers {
+
+    /**
+     * FIXME Originally used for OpportunityDataset upload, moved to SpatialLayers but should be named DataSource
```

Comment:
Should we rename this class?

Reply:
I have renamed it and done some other refactoring in 014db0689035d259d2443be86f8ee731ed87a75e.